mllib/src/main/scala/org/apache/spark/ml/param/shared/sharedParams.scala

```diff
@@ -88,6 +88,23 @@ private[ml] trait HasLabelCol extends Params {
 }
 
 /**
+ * Trait for shared param sampleWeightCol (default: "weight").
+ */
+private[ml] trait HasSampleWeightCol extends Params {
+
+  /**
+   * Param for sample weight column name.
+   * @group param
+   */
+  final val sampleWeightCol: Param[String] = new Param[String](this, "sampleWeightCol", "sample weight column name")
```

Comment:
Shall we call it `weightCol` and without default value? If this is defined, use the column specified as weights. Otherwise, treat all weights as 1.0.